core/trino-main/src/main/java/io/trino/execution/buffer/ClientBuffer.java

```diff
@@ -416,6 +415,12 @@ public void acknowledgePages(long sequenceId)
         dereferencePages(removedPages.build(), onPagesReleased);
     }
 
+    @SuppressWarnings("checkstyle:IllegalToken")
+    private void assertNotHoldsLock(String message)
+    {
+        assert !Thread.holdsLock(this) : message;
```

Comment:
That's against the project code style (as emphasized by the `@SuppressWarnings`).
Could you please supplement this place with some "excuse me" comment explaining why doing so?

Saying this is intentional sounds like a stupid thing to say -- i know it is intentional and you know that i know -- ... but a future reader may not know that we both knew.